lib/site-inspector.rb

```diff
@@ -127,41 +190,447 @@ def redirect
+      ) and
+      (
+        combos[:http][:root][:redirect_external] or
+        !combos[:http][:root][:up] or
+        combos[:http][:root][:status] >= 400
+      ) and
+      (
+        combos[:https][:www][:redirect_external] or
+        !combos[:https][:www][:up] or
+        combos[:https][:www][:https_bad_name] or
+        combos[:https][:www][:status] >= 400
+      ) and
+      (
+        combos[:https][:root][:redirect_external] or
+        !combos[:https][:root][:up] or
+        combos[:https][:root][:https_bad_name] or
+        combos[:https][:root][:status] >= 400
+      )
+    )
+
+    # OK, we've said a domain is a "redirect" domain.
+    # What does the domain redirect to?
+    if details[:redirect]
+      canon = combos[details[:canonical_protocol]][details[:canonical_endpoint]]
+      details[:redirect_to] = canon[:redirect_to]
+    else
+      details[:redirect_to] = nil
+    end
+
+    # HSTS on the canonical domain? (valid HTTPS checked in endpoint)
+    details[:hsts] = !!combos[:https][details[:canonical_endpoint]][:hsts]
+    details[:hsts_header] = combos[:https][details[:canonical_endpoint]][:hsts_header]
+
+    # HSTS on the entire domain?
+    details[:hsts_entire_domain] = !!(
+      combos[:https][:root][:hsts] and
+      combos[:https][:root][:hsts_details][:include_subdomains]
+    )
+
+    # HSTS preload-ready for the entire domain?
+    #
+    # Re-checks :hsts_entire_domain in case the :preload_ready
+    # flag ever changes its definition to not require include_subdomains.
+
+    details[:hsts_entire_domain_preload] = !!(
+      details[:hsts_entire_domain] and
+      combos[:https][:root][:hsts_details][:preload_ready]
+    )
+
+    details
   end
 
-  def to_hash
+  def endpoints
+    https_www = http_endpoint(true, true)
+    http_www = http_endpoint(false, true)
+    https_root = http_endpoint(true, false)
+    http_root = http_endpoint(false, false)
+
     {
-      :domain => domain.to_s,
-      :uri => uri.to_s,
-      :government => government?,
-      :live => !!response,
-      :ssl => https?,
-      :enforce_https => enforce_https?,
-      :non_www => non_www?,
-      :redirect => redirect,
-      :ip => ip,
-      :hostname => hostname.to_s,
-      :ipv6 => ipv6?,
-      :dnssec => dnssec?,
-      :cdn => cdn,
-      :google_apps => google_apps?,
-      :cloud_provider => cloud_provider,
-      :server => server,
-      :cms => cms,
-      :analytics => analytics,
-      :javascript => javascript,
-      :advertising => advertising,
-      :slash_data => slash_data?,
-      :slash_developer => slash_developer?,
-      :data_dot_json => data_dot_json?,
-      :click_jacking_protection => click_jacking_protection?,
-      :content_security_policy => content_security_policy?,
-      :xss_protection => xss_protection?,
-      :secure_cookies => secure_cookies?,
-      :strict_transport_security => strict_transport_security?,
-      :headers => headers
+      https: {
```

Comment:
This is a big change in behavior. Over in #22 we talked about having a hash and an expanded hash. This seems to essentially bury all the non-http checks (which would presumably be consistent between http and https).